packages/amplify_core/lib/src/types/query/query_field.dart

```diff
@@ -18,59 +18,185 @@ library query_field;
 import 'package:amplify_core/amplify_core.dart';
 import 'package:flutter/foundation.dart';
 
-import '../../util/parsers.dart';
-import '../models/model_field_type.dart';
-import '../temporal/datetime_parse.dart';
-
 part 'query_field_operators.dart';
 part 'query_pagination.dart';
 part 'query_predicate.dart';
 part 'query_sort.dart';
 
+/// {@template amplify_core.query_field}
+/// A field used to query models.
+///
+/// Contains methods for filtering and sorting query results.
+/// {@endtemplate}
 class QueryField<T> {
+  /// {@macro amplify_core.query_field}
+  const QueryField({required this.fieldName, this.fieldType});
+
+  /// The name of the filed that this is used to query.
```

Comment:
nit: typo, "field" ?

Reply:
Yeah, thanks for catching that. I'll update it